third_party/3/dateutil/parser.pyi

```diff
@@ -0,0 +1,52 @@
+from typing import List, Tuple, Optional, Callable
+from datetime import datetime
+
+__all__ = ...  # type: List[str]
+
+
+class parserinfo(object):
+    JUMP = ...  # type: List[str]
+    WEEKDAYS = ...  # type: List[Tuple[str, str]]
+    MONTHS = ...  # type: List[Tuple[str, str]]
+    HMS = ...  # type: List[Tuple[str, str, str]]
+    AMPM = ...  # type: List[Tuple[str, str, str]]
+    UTCZONE = ...  # type: List[str]
+    PERTAIN = ...  # type: List[str]
+    TZOFFSET = ...  # type: Dict[str, int]
+
+    def __init__(self, dayfirst: bool=..., yearfirst: bool=...) -> None: ...
+    def jump(self, name: str) -> bool: ...
+    def weekday(self, name: str) -> str: ...
+    def month(self, name: str) -> str: ...
+    def hms(self, name: str) -> str: ...
+    def ampm(self, name: str) -> str: ...
+    def pertain(self, name: str) -> bool: ...
+    def utczone(self, name: str) -> bool: ...
+    def tzoffset(self, name: str) -> int: ...
+    def convertyear(self, year: int) -> int: ...
+    def validate(self, year: datetime) -> bool: ...
+
+
+class parser(object):
+    def __init__(self, info: parserinfo=...) -> None: ...
+
+    def parse(
+        self,
+        timestr: str,
```

Comment:
I'm not overly familiar with how type annotations work, but you may want to be aware that `timestr` duck-types the input. In Python 3.x will take `bytes`, `str` or stream objects that implement `read()`. The preferred input is probably `str`, though.